rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -158,6 +173,11 @@ class Publisher : public PublisherBase
   virtual void
   publish(std::unique_ptr<MessageT, MessageDeleter> msg)
   {
+    if (std::is_same<MessageT, rcl_serialized_message_t>::value) {
+      this->template publish<MessageDeleter>(std::move(msg));
+      return;
+    }
```

Comment:
I think this call should not be possible if we keep `publish(const rcl_serialized_message_t * msg)` as mentioned below.